src/Desktop.Analyzers/Core/DoNotUseInsecureDTDProcessing.cs

```diff
@@ -0,0 +1,860 @@
+                        if (version != null)
+                        {
+                            context.RegisterCodeBlockStartAction<TLanguageKindEnum>(
+                                (c) =>
+                                {
+                                    RegisterAnalyzer(c, xmlTypes, version);
+                                });
+                        }
+                    }
+                });
+        }
+
+        private static bool ReferencesAnyTargetType(CompilationSecurityTypes types)
+        {
+            return types.XmlDocument != null
+                || types.XmlNode != null
+                || types.XmlReader != null
+                || types.XmlTextReader != null
+                || types.XPathDocument != null
+                || types.XmlSchema != null
+                || types.DataSet != null
+                || types.DataTable != null
+                || types.DataViewManager != null
+                || types.XmlSerializer != null;
+        }
+
+        private static DiagnosticDescriptor CreateDiagnosticDescriptor(LocalizableResourceString messageFormat, LocalizableResourceString description, string helpLink = null)
+        {
+            return new DiagnosticDescriptor(RuleId,
+                                            SecurityDiagnosticHelpers.GetLocalizableResourceString(nameof(DesktopAnalyzersResources.InsecureXmlDtdProcessing)),
+                                            messageFormat,
+                                            DiagnosticCategory.Security,
+                                            DiagnosticSeverity.Warning,
+                                            isEnabledByDefault: true,
+                                            description: description,
+                                            helpLinkUri: helpLink,
+                                            customTags: WellKnownDiagnosticTags.Telemetry);
+        }    
+
+        protected abstract void RegisterAnalyzer(CodeBlockStartAnalysisContext<TLanguageKindEnum> context, CompilationSecurityTypes types, Version targetFrameworkVersion);
+
+        protected abstract class Analyzer
+        {
+            // .NET frameworks >= 4.5.2 have secure default settings
+            private static readonly Version minSecureFxVersion = new Version(4, 5, 2);
+
+            private readonly CompilationSecurityTypes xmlTypes;
+            private readonly SyntaxNodeHelper syntaxNodeHelper;
+            private readonly bool isFrameworkSecure;
+
+            private readonly Dictionary<ISymbol, XmlDocumentEnvironment> xmlDocumentEnvironments = new Dictionary<ISymbol, XmlDocumentEnvironment>();
+            private readonly Dictionary<ISymbol, XmlTextReaderEnvironment> xmlTextReaderEnvironments = new Dictionary<ISymbol, XmlTextReaderEnvironment>();
+            private readonly Dictionary<ISymbol, XmlReaderSettingsEnvironment> xmlReaderSettingsEnvironments = new Dictionary<ISymbol, XmlReaderSettingsEnvironment>();
```

Comment:
That is correct, analyzers have always followed Roslyn "house style".

Sent from my Windows Phone
________________________________
From: Gen Lu<mailto:notifications@github.com>
Sent: ‎2/‎2/‎2016 6:12 PM
To: dotnet/roslyn-analyzers<mailto:roslyn-analyzers@noreply.github.com>
Cc: Larry Golding<mailto:Laurence.Golding@microsoft.com>
Subject: Re: [roslyn-analyzers] Add XML rules CA3075 Only (#705)


In src/Desktop.Analyzers/Core/DoNotUseInsecureDTDProcessing.cs<https://github.com/dotnet/roslyn-analyzers/pull/705#discussion_r51669967>:

> +        }
> +
> +        protected abstract void RegisterAnalyzer(CodeBlockStartAnalysisContext<TLanguageKindEnum> context, CompilationSecurityTypes types, Version targetFrameworkVersion);
> +
> +        protected abstract class Analyzer
> +        {
> +            // .NET frameworks >= 4.5.2 have secure default settings
> +            private static readonly Version minSecureFxVersion = new Version(4, 5, 2);
> +
> +            private readonly CompilationSecurityTypes xmlTypes;
> +            private readonly SyntaxNodeHelper syntaxNodeHelper;
> +            private readonly bool isFrameworkSecure;
> +
> +            private readonly Dictionary<ISymbol, XmlDocumentEnvironment> xmlDocumentEnvironments = new Dictionary<ISymbol, XmlDocumentEnvironment>();
> +            private readonly Dictionary<ISymbol, XmlTextReaderEnvironment> xmlTextReaderEnvironments = new Dictionary<ISymbol, XmlTextReaderEnvironment>();
> +            private readonly Dictionary<ISymbol, XmlReaderSettingsEnvironment> xmlReaderSettingsEnvironments = new Dictionary<ISymbol, XmlReaderSettingsEnvironment>();


Consider analyzers used to be part of roslyn repo, I'd say it's the case from the very begining. I probably didn't pay enough attention last time, my bad :) By the way, I will fix those already merged in when I get time, so don't worry about it.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/roslyn-analyzers/pull/705/files#r51669967>.